ICO-2939/JSON-logging-framework

josippapez opened the pull request:
Added GlookoLogging module for logging messages.
Wrote test for GlookoLogging module.
Minor rubocop fixes.

**Format of the message Is viable to change (include/exclude certain parts of the message).**

josippapez committed:
ICO-2939/JSON-logging-framework

Added GlookoLogging module for logging messages.
Wrote test for GlookoLogging module.
Minor rubocop fixes.

VlatkoGoljevacki commented on spec/lib/base/glooko_logging_spec.rb:
```diff
@@ -0,0 +1,29 @@
+  let(:logger_class) { Class.new { include Base::GlookoLogging } }
+
+  it 'logs expected output with json data' do
+    test = expect do
```
can we write this using lazy variables?

let(:test_log) do 
  logger_class.new.log('warn', 
                                      {fog: 'test'}, 
                                      __LINE__, 
                                      __method__,
                                       __FILE__) 
end

expect(test_log).to output()...

josippapez replied:
Changed the code to fit the suggested comment

VlatkoGoljevacki commented on spec/lib/base/glooko_logging_spec.rb:
```diff
@@ -0,0 +1,29 @@
+
+  it 'logs expected output with json data' do
+    test = expect do
+      logger_class.new.log('warn', {fog: 'test'}, __LINE__, __method__,
```
can we also test that the Logger received "warn" when logging the data?

josippapez replied:
Please check if the latest change includes the requested test

josippapez committed:
ICO-2939/JSON-logging-framework

Wrote test for checking if correct severity is displayed.
Moved log method to lazy variable.

josippapez committed:
ICO-2939/JSON-logging-framework

Fixed typo

josippapez committed:
ICO-2939/JSON-logging-framework

Fixed arguments test for glooko logging.

elliothutchinson reviewed (commented):
I see this isn't in the AC for the jira, but is the work left for a future version to get things like glooko_code, request_id, etc out of the context/thread local?

josippapez commented:
> I see this isn't in the AC for the jira, but is the work left for a future version to get things like glooko_code, request_id, etc out of the context/thread local?

Yeah, I was thinking of just setting up basic logger for now and when we have finalized log format, we can add/remove data.

VlatkoGoljevacki commented:
> > I see this isn't in the AC for the jira, but is the work left for a future version to get things like glooko_code, request_id, etc out of the context/thread local?
> 
> Yeah, I was thinking of just setting up basic logger for now and when we have finalized log format, we can add/remove data.

@josippapez Can you create a ticket for this and put in the prioritized backlog so we don't forget about it this sprint?

josippapez commented:
@VlatkoGoljevacki yes, I'll add it to the backlog
